### DISCUSSION TITLE
Is there a way to use pocketbase with prisma?

### DISCUSSION THREAD

**Discussion by valerius21** [pocketbase/pocketbase#465] [Q&A] [answered]
Hi there,

I had lots of success with projects using Prisma in the past. I know that there is a JS/TS library to query the database records. 
However, since Prisma supports SQLite and, as far as I can tell, pocketbase makes some use of SQLite I was wondering if there is a way to access the database via. prisma.

So far connecting to the `pb_data/data.db` file resulted in connection erros.

**Comment by ganigeorgiev** [comment, marked as answer]
Nothing is stopping you to load/connect the pocketbase db file in Prisma. The error you've stumbled on is probably because of a read/write lock or something similar.

But what is the point of using PocketBase in the first place if you are going to query the database with Prisma or any other external orm?

You want to use PocketBase because of the abstractions that it provide - you have pagination, sorting, validations and **most importantly acces checks** for your data. It's not just db read/write.

**Reply by valerius21** [reply]
thank you for your insight! I already started writing most of my project with Prisma, but your statement sounds compelling. I'll try the JS SDK out. :D

**Reply by devidw** [reply]
> But what is the point of using PocketBase in the first place if you are going to query the database with Prisma or any other external orm?

One might want to leverage Prisma's type safe db queries on the backend and use pocketbase for things like authentication and storage.

**Reply by ganigeorgiev** [reply]
@devidw This is a very old discussion but my other replies remain valid and I don't recommend using PocketBase if you want to use Prisma. If you haven't already, please read also https://pocketbase.io/docs/how-to-use/

**Comment by bn-l** [comment]
I found this after googling. My own purpose would be using prisma's schema DSL which is very nice.

**Reply by ganigeorgiev** [reply]
You wouldn't benefit much from doing that. If all you want is to use SQLite, you don't need PocketBase, just use Prisma (if you are considering combining it with a JS meta-framework, please read https://github.com/pocketbase/pocketbase/discussions/5313).

**Reply by bn-l** [reply]
Thanks for the reply. It’s more because I find it easier to reason about with their schema language and I like that I can also put it into source control. As to why not just prisma: I want the real time SSR updates, the JS SDK API, the go hooks and the client side access (as an alternative to fire base). (This is not a covert feature request fyi).

**Reply by ganigeorgiev** [reply]
We have migrations for this. Please read the documentation and more specifically:
- https://pocketbase.io/docs/js-migrations/
- https://pocketbase.io/docs/js-collections/

Changing the schema externally could cause all sort of problems. PocketBase has its own specific way to define and structure the table fields and if you are not using the Collection model it will most likely result in a failure.
